front-end/src/components/analysis-page/IntentDiagram.jsx

```diff
@@ -98,7 +98,7 @@ class IntentDiagram extends Component {
                         </label>
                     </button>
                 </div>
-                <div className="rounded-lg flex flex-col gap-y-3 md:flex-row mx-auto justify-between text-center gap-x-9 m-9">
+                <div className="rounded-lg flex flex-col gap-y-3 md:flex-row mx-auto justify-evenly text-center overflow-auto gap-x-5 m-9 sm:min-w-[37rem] sm:max-w-[35rem] 2xl:max-w-[50rem] 2xl:max-h-[14rem]">
```

Comment:
```suggestion
                <div className="rounded-lg flex flex-col gap-y-3 md:flex-row mx-auto md:justify-evenly text-center md:overflow-auto gap-x-5 m-9 sm:min-w-[37rem] sm:max-w-[35rem] 2xl:max-w-[50rem] 2xl:max-h-[14rem]">
```
or else we get
![image](https://user-images.githubusercontent.com/76976842/205066276-92200b33-2b92-4a5e-ab30-28c06fb14ef6.png)